### ISSUE TITLE
[Enhancement] Display cluster-level information in the ReactFlow workspace

### ISSUE THREAD

**Issue by ohltyler** [opensearch-project/dashboards-flow-framework#377]
The current view of ingest and search pipelines in the ReactFlow workspace is ambiguous on where/when these resources are created. A few ways this can be improved.

1. (first goal) _Where_ the resources are created. We should display the cluster (local or remote) within the workspace in a visual way. Perhaps a larger parent node encapsulating both the ingest and search flows (see example below). At the least, have more helpful text indicating that these pipelines are being built within the specified cluster.

<img width="699" alt="Screenshot 2024-09-13 at 1 50 15 PM" src="https://github.com/user-attachments/assets/1d83ea86-7294-4b96-ab86-55160849dfe4">

2. (stretch goal): _When_ the resources are created. Users will iteratively build out these pipelines, and we should reflect what is created only after the users create them, not just based on the UI configuration. This could mean showing the pipelines as greyed-out/disabled, or not shown at all, until they are concretely provisioned.

Example:

<img width="704" alt="Screenshot 2024-09-13 at 1 56 45 PM" src="https://github.com/user-attachments/assets/6cd6cc91-01e8-4bed-becf-9b6bfbb723eb">

**Comment by dblock** [comment]
[Catch All [Triage](https://github.com/opensearch-project/.github/blob/main/TRIAGING.md) - [1](https://github.com/dblock), [2](https://github.com/reta),  [3](https://github.com/krisfreedain),  [4](https://github.com/lukas-vlcek/)]

**Comment by ohltyler** [comment]
Another potential option for Item 2 is adding labeling/status per pipeline (or per-component) indicating if it has been provisioned or not. Overall, the "Preview" is supposed to show the proposed configuration, and does not need to be one-to-one parity with what is provisioned.

**Comment by kamingleung** [comment]
@ohltyler What are the statuses and states we have for Ingest/Search pipeline? Do the two pipelines have different states?

**Comment by ohltyler** [comment]
For each pipeline individually, we currently track:
1. If it has been provisioned or not
2. If the form differs than what's been provisioned (unsaved "provision" changes - e.g., configure and provision processor A, configure processor B, but not yet provisioned processor B)

Further per-processor granularity is available for ingest, and for search, there is ongoing work for that - see https://github.com/opensearch-project/OpenSearch/issues/14745. I'd consider per-processor state a P2+-type item based on the backend dependencies.

**Comment by kamingleung** [comment]
For First goal, we can leverage the eixsting multiple data source picker to display which cluster this is belonging to.

**Comment by kamingleung** [comment]
For the Second goal, we can consider displaying statuses to the pipelines. 

- One area to display this is in the preview of a pipeline like this with [OuiHealth](https://oui.opensearch.org/1.17/#/display/health):
![image](https://github.com/user-attachments/assets/d95f8c08-3fb4-47f8-892a-8db2fc98adbb)
We should 

- We can also consider displaying the statuses in the resources lists: we can add a `Status` column to the resource table:
![image](https://github.com/user-attachments/assets/5eddc336-3d1b-4a9d-a7dc-8583b7eec06c)

**Comment by kamingleung** [comment]
To help exploring the states, I am mapping out the lifecycle of the pipeline as users are working through the workflows:
![image](https://github.com/user-attachments/assets/96ab8380-dda5-438e-b05a-1a90aa3fdf36)

Proposing:
**Pipeline status:**
- `Inactive`: The pipeline has not been created.
- `Provisioning`: When the pipeline is building or updating. (Are we able to get this state? How long does it usually take to build?)
- `Active`: The pipeline is running.

**Save status**
- `Pending changes`: Perhaps we can show a secondary status for tracking changes that has not been provisioned yet.

**Comment by ohltyler** [comment]
Thanks Ming. Regarding the ‘Provisioning’ state, it will be quite immediate, in my opinion I don’t think we want to / should show. It would just flash for a moment and could cause user confusion. The rest of those states make sense to me.

From: "Ka Ming L." ***@***.***>
Reply-To: opensearch-project/dashboards-flow-framework ***@***.***>
Date: Thursday, November 7, 2024 at 7:16 PM
To: opensearch-project/dashboards-flow-framework ***@***.***>
Cc: "Ohlsen, Tyler" ***@***.***>, Mention ***@***.***>
Subject: Re: [opensearch-project/dashboards-flow-framework] [Enhancement] Display cluster-level information in the ReactFlow workspace (Issue #377)


To help exploring the states, I am mapping out the lifecycle of the pipeline as users are working through the workflows:
image.png (view on web)<https://github.com/user-attachments/assets/96ab8380-dda5-438e-b05a-1a90aa3fdf36>

Proposing:
Pipeline status:

  *   Inactive: The pipeline has not been created.
  *   Provisioning: When the pipeline is building or updating. (Are we able to get this state? How long does it usually take to build?)
  *   Active: The pipeline is running.

Save status

  *   Pending changes: Perhaps we can show a secondary status for tracking changes that has not been provisioned yet.

**Comment by kamingleung** [comment]
@ohltyler Got it. We may want to consider having this "Provisioning" state in the future to cover edge cases that requires more time.

**Comment by ohltyler** [comment]
Several UX enhancements to indicate created resources and how to access them have been completed. Marking this issue as irrelevant now.
